cmd/snap/complete.go

```diff
@@ -28,9 +31,38 @@ func (s installedSnapName) Complete(match string) []flags.Completion {
 	return ret
 }
 
+func completeFromSortedFile(fn, match string) ([]flags.Completion, error) {
+	fd, err := os.Open(fn)
+	if err != nil {
+		return nil, err
+	}
+	defer fd.Close()
+
+	var ret []flags.Completion
+
+	// TODO: the file is sorted; we should bisect it
```

Comment:
How much data are you expecting to see in the particular use case that you wrote this for?

Reply:
I hope it'll grow to be in the millions. I expect it to be in the hundreds of thousands at least.